### PR TITLE
Add MergeTreeSettings to system.settings_changes

### LINKED ISSUE #72221: Control MergeTreeSettings changes via CI
We introduced merge_tree_settings_changes_history in SettingsChangesHistory. It's not controlled by the CI, ergo not respected which means we might break compatibility and won't notice it automatically.

### PR DESCRIPTION
**Pull request by Algunenano**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add MergeTreeSettings to system.settings_changes

It also add checks to CI and controls changes in both stateless and upgrade checks.

Closes https://github.com/ClickHouse/ClickHouse/issues/72221

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Run only fuzzers related jobs (libFuzzer fuzzers, AST fuzzers, etc.)
- [ ]  Exclude: AST fuzzers
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by robot-ch-test-poll2** [comment]
*This is an automated comment for commit b0d555b892351878999f1dafcc86e52f59953e8a with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/72694/b0d555b892351878999f1dafcc86e52f59953e8a/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/b81ee27750c4520de6e6b979ce7d4f6a111cb8ea/docs_check.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

File: src/Core/SettingsChangesHistory.cpp [hunk 1]
```diff
@@ -616,7 +616,9 @@ static std::initializer_list<std::pair<ClickHouseVersion, SettingsChangesHistory
 {
     {"24.12",
         {
-            {"enforce_index_structure_match_on_partition_manipulation", true, false, "Add new setting to allow attach when source table's projections and secondary indices is a subset of those in the target table."}
+            {"enforce_index_structure_match_on_partition_manipulation", true, false, "Add new setting to allow attach when source table's projections and secondary indices is a subset of those in the target table."},
+            {"use_primary_key_cache", false, false, "New setting"},
+            {"prewarm_primary_key_cache", false, false, "New setting"},
```
Comment (novikd) [inline src/Core/SettingsChangesHistory.cpp:621]: Why? It doesn't seem to be related

File: src/Storages/System/StorageSystemSettingsChanges.cpp [hunk 2]
```diff
@@ -12,6 +12,7 @@ ColumnsDescription StorageSystemSettingsChanges::getColumnsDescription()
     /// TODO: Fill in all the comments
     return ColumnsDescription
     {
+        {"type", std::make_shared<DataTypeString>(), "The group of settings (Core, MergeTree...)"},
```
Comment (novikd) [inline src/Storages/System/StorageSystemSettingsChanges.cpp:15]: Maybe better to make it `Enum`?

File: src/Core/SettingsChangesHistory.cpp (hunk 1 above)
Comment (Algunenano) [reply src/Core/SettingsChangesHistory.cpp:621]: They were introduced in current master but not added to the history
